garrysmod/lua/includes/modules/team.lua

```diff
@@ -170,14 +183,18 @@ end
 
 function GetColor( index )
 
-	if (!TeamInfo[index]) then return DefaultColor end
-	return table.Copy( TeamInfo[index].Color )
+	if ( !TeamInfo[index] ) then return DefaultColor end
+
+	local color = TeamInfo[index].Color
+	if ( color == nil ) then return nil end
+
+	return setmetatable( { r = color.r, g = color.g, b = color.b, a = color.a }, getmetatable(color) )

```

Comment:
This should just use the `Color` global function. It may be slower, but it at least doesn't duplicate its internals unnecessarily.